compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3054,7 +3054,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             let (span, separator) = if let [.., last] = bounds {
                                 (last.span().shrink_to_hi(), " +")
                             } else {
-                                (ident.span.shrink_to_hi(), ":")
+                                (generics.span.shrink_to_hi(), ":")
```

Comment:
Right, and if there are no generic params, this is still correct since the span will be a zero-sized span directly after the ident :+1: